File: services/crud/src/main/java/com/blog/crud/likes/Like.java
package com.blog.crud.likes;

import com.blog.crud.blog.Blog;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
@Entity(name = "_like")
public class Like {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "like_seq")
    @SequenceGenerator(name = "like_seq", sequenceName = "like_sequence", allocationSize = 50)
    private Integer Id;

    @Enumerated(EnumType.STRING)
    private LikeType likeType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "blog_id")
    private Blog blog;

    private String userId;

    @CreatedDate
    @Column(updatable = false, nullable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(insertable = false)
    private LocalDateTime lastModifiedAt;
}

File: services/feed/models/like.go
package models

import "time"

type Like struct {
	ID             int        `db:"id"`
	LikeType       string     `db:"like_type"`
	BlogID         int        `db:"blog_id"`
	UserID         string     `db:"user_id"`
	CategoryID     int        `db:"category_id"`
	CreatedAt      *time.Time `db:"created_at"`
	LastModifiedAT *time.Time `db:"last_modified_at"`
}
